internal/core/application/operator_service.go

```diff
@@ -1656,21 +1664,79 @@ func (o *operatorService) GetMarketReport(
 
 			volume[swapRequest.GetAssetR()] += int64(swapRequest.GetAmountR())
 			volume[swapRequest.GetAssetP()] += int64(swapRequest.GetAmountP())
+
+			for i, v := range groupedVolume {
+				//find time slot to which trade belongs to and calculate volume for that slot
+				if (time.Unix(int64(trade.SwapRequest.Timestamp), 0).After(v.StartTime) ||
+					time.Unix(int64(trade.SwapRequest.Timestamp), 0).Equal(v.StartTime)) &&
+					(time.Unix(int64(trade.SwapRequest.Timestamp), 0).Before(v.EndTime) ||
+						time.Unix(int64(trade.SwapRequest.Timestamp), 0).Equal(v.EndTime)) {
+
+					//assume AmountR is base asset, AmountP(FeeAsset) is quote asset
+					volumeBaseAmount := swapRequest.GetAmountR() + v.BaseVolume
+					volumeQuoteAmount := swapRequest.GetAmountP() + v.QuoteVolume
+					if swapRequest.GetAssetR() == market.QuoteAsset {
+						volumeBaseAmount = swapRequest.GetAmountP() + v.BaseVolume
+						volumeQuoteAmount = swapRequest.GetAmountR() + v.QuoteVolume
+					}
+
+					groupedVolume[i] = MarketVolume{
+						BaseVolume:  volumeBaseAmount,
+						QuoteVolume: volumeQuoteAmount,
+						StartTime:   v.StartTime,
+						EndTime:     v.EndTime,
```

Comment:
shall we [Time.MarshalText](https://pkg.go.dev/time#Time.MarshalText) as suggested by @Janaka-Steph  to "beautify" the string?